### DISCUSSION TITLE
onProgress alternatives

### DISCUSSION THREAD

**Discussion by vortigont** [vortigont/esp32-flashz#7] [Ideas]
### 👋 Let's discuss the idea of getting some progress update while writing compressed data
Continued from #6 

So there are few things that should be considered
  * Arduino's UpdateClass has no virtual methods, so I can't override any of it to inject different logic there, only new methods could be used
  * when writing compressed data, the size of resulting uncompressed bulk writes is unknown until last block is decompressed
  * when starting compressed writes parent UpdateClass gets target size as `UPDATE_SIZE_UNKNOWN`, that internally is being translated to partition size. This makes progress calculation quite tricky

To summarize, the are only two things that could be known for sure when doing progress callbacks with compressed writes
 * current size of compressed data processed
 * current size of uncompressed inflated data so far

**Comment by Mr-HaleYa** [comment]
Without knowing the final size there's not much that can be done.
Could set an after-x-amount-of-bytes-processed callback, which may be better than nothing.
This could be used when knowing the total compressed size to get an approximation.
